### PR TITLE
Change and create system classes for support Mitaka Ceilometer

### PR DESCRIPTION
**Pull request by ityaptin**
Aodh and Ceilometer system classes were created in this commit.
Also keystone, galera and mysql classes were changed to support
aodh installation.

### CODE REVIEW

File: classes/system/ceilometer/server/cluster.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  _param:
+    ceilometer_influxdb_user: ceilometer
+    ceilometer_influxdb_password: ${_param:stacklight_influxdb_password}
+    ceilometer_influxdb_database: ceilometer
+    ceilometer_influxdb_host: ${_param:monitoring_service_address}
+    ceilometer_influxdb_port: 8086
+    ceilometer_elasticsearch_enable: true
+    ceilometer_elasticsearch_port: 9200
+    ceilometer_elasticsearch_host: ${_param:monitoring_service_address}
+  haproxy:
```
Comment (cznewt) [inline classes/system/ceilometer/server/cluster.yml]: please split haproxy listens to system.haproxy.....

Comment (ityaptin) [inline classes/system/ceilometer/server/cluster.yml]: Done

File: classes/system/aodh/server/cluster.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+classes:
+- service.aodh.server.cluster
+- service.haproxy.proxy.single
+- service.keepalived.cluster.single
+parameters:
```
Comment (simonpasquier) [inline classes/system/aodh/server/cluster.yml]: please remove this dict

File: classes/system/keystone/client/single.yml [hunk 3]
```diff
@@ -1,4 +1,6 @@
 classes:
+- system.keystone.client.service.aodh
+- system.keystone.client.service.ceilometer
```
Comment (simonpasquier) [inline classes/system/keystone/client/single.yml]: these 2 services shouldn't be included by default

File: classes/system/heka/ceilometer_collector/single.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      influxdb_port: 8086
+      influxdb_username: ceilometer
+      resource_decoding: false
+      rabbit_host: ${_param:cluster_vip_address}
+      rabbit_port: 5672
+      rabbit_user: openstack
+      rabbit_password: ${_param:rabbitmq_openstack_password}
+      rabbit_vhost: '/openstack'
+      rabbit_queue: metering.sample
+    _support:
```
Comment (simonpasquier) [inline classes/system/heka/ceilometer_collector/single.yml]: to be removed

File: classes/system/galera/server/database/aodh.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+parameters:
```
Comment (simonpasquier) [inline classes/system/galera/server/database/aodh.yml]: I'm confused which of this class and system.mysql.client.database.aodh is used...

File: classes/system/aodh/server/cluster.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  aodh:
+    server:
+      logging:
+        heka:
+          enabled: true
```
Comment (elemoine) [inline classes/system/aodh/server/cluster.yml]: Isn't logging:heka:enabled something from the past?

File: classes/system/galera/server/database/aodh.yml (hunk 5 above)
Comment (ityaptin) [inline classes/system/galera/server/database/aodh.yml]: How I understand by https://github.com/tcpcloud/salt-formula-mysql/blob/master/mysql/client/server.sls#L4
parameter mysql.server.database is used, not mysql.client.server.database.database from system.mysql.client.database.aodh. But I made it by analogy with mysql formulas of keystone service for the possibility that I missed smth.

File: classes/system/keystone/client/single.yml (hunk 3 above)
Comment (ityaptin) [inline classes/system/keystone/client/single.yml]: Done

File: classes/system/aodh/server/cluster.yml (hunk 2 above)
Comment (ityaptin) [inline classes/system/aodh/server/cluster.yml]: done

File: classes/system/aodh/server/cluster.yml (hunk 6 above)
Comment (ityaptin) [inline classes/system/aodh/server/cluster.yml]: You are completely right. I deleted in in last version

File: classes/system/ceilometer/agent/single.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+classes:
+- service.ceilometer.agent.cluster
+- service.ceilometer.agent.publisher.graphite
+parameters:
+  server:
+    logging:
+      heka:
+        enabled: true
```
Comment (elemoine) [inline classes/system/ceilometer/agent/single.yml]: To be removed.

File: classes/system/ceilometer/server/cluster.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    ceilometer_influxdb_host: ${_param:monitoring_service_address}
+    ceilometer_influxdb_port: 8086
+    ceilometer_elasticsearch_enable: true
+    ceilometer_elasticsearch_port: 9200
+    ceilometer_elasticsearch_host: ${_param:monitoring_service_address}
+  ceilometer:
+    server:
+      logging:
+        heka:
+          enabled: true
```
Comment (elemoine) [inline classes/system/ceilometer/server/cluster.yml]: To be removed.

Comment (ityaptin) [inline classes/system/ceilometer/server/cluster.yml]: Done

File: classes/system/ceilometer/agent/single.yml (hunk 7 above)
Comment (ityaptin) [inline classes/system/ceilometer/agent/single.yml]: Done

File: classes/system/ceilometer/server/cluster.yml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+parameters:
+  _param:
+    ceilometer_influxdb_user: ceilometer
+    ceilometer_influxdb_password: ${_param:stacklight_influxdb_password}
+    ceilometer_influxdb_database: ceilometer
+    ceilometer_influxdb_host: ${_param:monitoring_service_address}
+    ceilometer_influxdb_port: 8086
+    ceilometer_elasticsearch_enable: true
+    ceilometer_elasticsearch_port: 9200
+    ceilometer_elasticsearch_host: ${_param:monitoring_service_address}
```
Comment (elemoine) [inline classes/system/ceilometer/server/cluster.yml]: Having all these params here in classes/system/ceilometer suggest that all the clusters that include Ceilometer will use Ceilometer with the InfluxDB and Elasticsearch plugins. That sounds reasonable to me, but I just wanted to make sure that's what you had in mind as well.

File: classes/system/haproxy/proxy/listen/openstack/aodh.yml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+parameters:
+  haproxy:
+    proxy:
+      listen:
+        aodh-api:
+          type: openstack-service
```
Comment (elemoine) [inline classes/system/haproxy/proxy/listen/openstack/aodh.yml]: Please add `service_name: aodh`.

File: classes/system/haproxy/proxy/listen/openstack/ceilometer.yml [hunk 11]
```diff
@@ -0,0 +1,22 @@
+parameters:
+  haproxy:
+    proxy:
+      listen:
+        ceilometer-api:
+          type: openstack-service
```
Comment (elemoine) [inline classes/system/haproxy/proxy/listen/openstack/ceilometer.yml]: Please add `service_name: ceilometer`.

File: classes/system/haproxy/proxy/listen/openstack/aodh.yml (hunk 10 above)
Comment (ityaptin) [inline classes/system/haproxy/proxy/listen/openstack/aodh.yml]: Done

File: classes/system/haproxy/proxy/listen/openstack/ceilometer.yml (hunk 11 above)
Comment (ityaptin) [inline classes/system/haproxy/proxy/listen/openstack/ceilometer.yml]: Done

File: classes/system/galera/server/database/aodh.yml (hunk 5 above)
Comment (simonpasquier) [inline classes/system/galera/server/database/aodh.yml]: @ityaptin, see this commit where it only adds to classes.system.galera.server.database... I would remove the system.mysql.client.database.aodh class for clarity.

File: classes/system/ceilometer/agent/single.yml [hunk 12]
```diff
@@ -0,0 +1,3 @@
+classes:
+- service.ceilometer.agent.single
+- service.ceilometer.agent.publisher.graphite
```
Comment (cznewt) [inline classes/system/ceilometer/agent/single.yml]: get rid of the graphite publisher

Comment (ityaptin) [inline classes/system/ceilometer/agent/single.yml]: done

Comment (ityaptin) [inline classes/system/ceilometer/agent/single.yml]: @cznewt done

**Pull request merged**